### ISSUE TITLE
fix(common): don't capitalize letter after a quote in TitleCasePipe

### ISSUE THREAD

**Issue by oreofeolurin** [angular/angular#16586]
**I'm submitting a ...**  (check one with "x")
```
[*] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
The TitleCasePipe capitalizes letters after a quote. Quote seems to be a break.

For example: `{{"it's complicated" | titlecase}}` produces the output `It'S Complicated`.

**Expected behavior**
The letter after the quotes should not be capitalized.

Using the same example as above: `{{"it's complicated" | titlecase}}`  produces the output `It's Complicated`

**What is the motivation / use case for changing the behavior?**
Intend to change the title case of words properly

**Please tell us about your environment:**
Windows 10

* **Angular version:** 2.0.X
4.0.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
all

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

* **Node (for AoT issues):** `node --version` =   
6.9.1

**Comment by KyleCartmell** [comment]
https://github.com/angular/angular/blob/master/packages/common/src/pipes/case_conversion_pipes.ts
Looks like the regex splitting words is using the Javascript "\b" word boundary identifier. This is also going to happen around accented characters.

**Comment by Shne** [comment]
This is also broken for common danish characters æøå.

e.g. "føderation" => "FØDeration". "jordbær" => "JordbÆR"

**Comment by christopher-harris** [comment]
I have seen this as well.

**Comment by krzSobin** [comment]
There is similar error in Angular 5.0.0 as well.

`<h3 class="panel-title">{{dailyMeals.date | date: 'EEEE, dd/MM/yyyy':'':'pl' | titlecase }}</h3>`
produces `PoniedziaŁEk, 25/12/2017`
(Poniedziałek is Monday in Polish)

And it should produce `Poniedziałek, 25/12/2017`

**Comment by Korouzan** [comment]
Get the same issue with french local.
{{"décembre" | titlecase}} or {{myDate | date :'MMM':'fr-FR' | titlecase}} return "DÉCembre"

**Comment by vicb** [comment]
I have just closed https://github.com/angular/angular/pull/20123 because the implementation was not really helping.

Seems like a quick google serach returns interesting results:
- https://www.codetuts.tech/title-case-capitalize-sentence/
- [voca.js](https://github.com/panzerdp/voca/blob/master/src/helper/reg_exp/const_extended.js)
- ...

It should be easy to improve the current pipe for someone that want to spend a couple hours looking at the existing impls. Any help here would be appreciated.

**Comment by linflux** [comment]
I still have the same issue with portuguese Brazil:
if viewFirstMonth date is March (Março, in locale 'pt') the pipe transforms to MarÇO
{{ viewFirstMonth | date:'MMMM' | titlecase }}

**Comment by sergeome** [comment]
@vicb,

I created a PR with the fix for this issue. As @Shne suggested, I took already existing implementation and added some unit tests to verify broken functionality.

**Comment by gkalpak** [comment]
Fixed by #22600.
